pandas/core/frame.py

```diff
@@ -5177,53 +5177,55 @@ def append(self, other, ignore_index=False, verify_integrity=False):
     def join(self, other, on=None, how='left', lsuffix='', rsuffix='',
              sort=False):
         """
-        Join columns with other DataFrame either on index or on a key
-        column. Efficiently Join multiple DataFrame objects by index at once by
+        Join columns with other DataFrame either on index or a key
+        column.
+
+        Efficiently Join multiple DataFrame objects by index at once by
         passing a list.
 
         Parameters
         ----------
         other : DataFrame, Series with name field set, or list of DataFrame
             Index should be similar to one of the columns in this one. If a
             Series is passed, its name attribute must be set, and that will be
-            used as the column name in the resulting joined DataFrame
+            used as the column name in the resulting joined DataFrame.
         on : name, tuple/list of names, or array-like
```

Comment:
The types could use a clean up here. Something like:

`str, list of str, tuple of (str,), or array-like`